site/documentation/content/api/telemetry-pubsub/index.md

```diff
@@ -0,0 +1,118 @@
+this page we will simply use *Pub/Sub* when referring to Google Pub/Sub.
+
+The Telemetry API for Pub/Sub is an alternative to the [Telemetry API for Kafka]({{< relref "/api/telemetry-kafka" >}})
+for applications that want to consume telemetry data from a Pub/Sub Messaging Network instead of an Apache Kafka&trade;
+broker.
+
+## Southbound Operations
+
+The following operation can be used by *Protocol Adapters* to publish telemetry data received from devices to
+consumers like *Business Applications*.
+
+### Publish Telemetry Data
+
+The protocol adapter writes messages to the tenant-specific topic `projects/${google_project_id}/topics/${tenant_id}.telemetry`
+where `${google_project_id}` is the ID of the Google project and `${tenant_id}` is the
+ID of the tenant that the client wants to upload telemetry data for.
+
+**Preconditions**
+
+1. The Google Project ID is declared in the deployment file.
+1. The client is authorized to write to the topic.
+1. The device for which the adapter wants to send telemetry data has been registered (see
+   [Device Registration API]({{< relref "/api/device-registration" >}})).
+
+**Message Flow**
+
+It is up to the discretion of the protocol adapter whether it wants to use *AT LEAST ONCE* or *AT MOST ONCE* delivery
+semantics.
+
+The following sequence diagram illustrates the flow of messages involved in the *HTTP Adapter* publishing a telemetry
+data message to the Pub/Sub API implementing *AT MOST ONCE* delivery semantics.
+
+{{< figure src="publish_pubsub_qos0.svg" title="Publish telemetry data flow (AT MOST ONCE)" >}}
+
+1. *Device* `4711` PUTs telemetry data to the *HTTP Adapter*
+    1. *HTTP Adapter* publishes telemetry data to the *Pub/Sub API*.
+    1. *HTTP Adapter* acknowledges the reception of the data to the *Device*.
+
+The following sequence diagram illustrates the flow of messages involved in the *HTTP Adapter* publishing a telemetry
+data message to the Pub/Sub API implementing *AT LEAST ONCE* delivery semantics.
+
+{{< figure src="publish_pubsub_qos1.svg" title="Publish telemetry data flow (AT LEAST ONCE)" >}}
+
+1. *Device* `4711` PUTs telemetry data to the *HTTP Adapter*, indicating *QoS Level* 1.
+    1. *HTTP Adapter* publishes telemetry data to the *Pub/Sub API*.
+    1. *Pub/Sub* acknowledges reception of the message.
+    1. *HTTP Adapter* acknowledges the reception of the data to the *Device*.
+
+When a Pub/Sub publisher raises an exception while sending a telemetry message to Pub/Sub, the protocol adapter MUST NOT try
+to re-send such rejected messages but SHOULD indicate the failed transfer to the device if the transport protocol
+provides means to do so.
+
+**Message Format**
+
+The following tables provides an overview of the message format for a PubsubMessage. The message must contain either a non-empty data field or at least one attribute.
```

Comment:
> ... tables provide ...

> ... PubSub message...

Does this mean that a message can not contain both data and attributes?

Reply:
A message can contain both, but at least one of them must be set on a message.

Reply:
Then IMHO it should be rephrased to something like
> The message must contain a non-empty data field or at least one attribute (or both).

to make this more explicit.

Reply:
Agree, rephrased it to your suggestion